pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -73,959 +71,762 @@
+                    torch.device("cuda", i) for i in device_parser.parse_gpu_ids(self._device_flag)
+                ]
+            else:
+                self._parallel_devices = [torch.device("cuda", i) for i in self._device_flag]
+
+        elif self._accelerator_flag == "cpu":
+            self.accelerator = CPUAccelerator()
+            if self._device_flag == "auto" or not self._device_flag:
+                self._device_flag = CPUAccelerator.auto_device_count()
+            if isinstance(self._device_flag, int):
+                self._parallel_devices = [torch.device("cpu")] * self._device_flag
+            else:
+                rank_zero_warn(
                     "The flag `devices` must be an int with `accelerator='cpu'`,"
-                    f" got `devices={self.devices}` instead."
+                    f" got `devices={self._device_flag}` instead."
                 )
-            self.num_processes = self.devices
-            return True
-        return False
 
-    @property
-    def use_dp(self) -> bool:
-        return self._strategy_type == _StrategyType.DP
+        self._gpus = self._device_flag if not self._gpus else self._gpus
 
-    @property
-    def use_ddp(self) -> bool:
-        return self._strategy_type in (
-            _StrategyType.DDP,
-            _StrategyType.DDP_SPAWN,
-            _StrategyType.DDP_SHARDED,
-            _StrategyType.DDP_SHARDED_SPAWN,
-            _StrategyType.DDP_FULLY_SHARDED,
-            _StrategyType.DEEPSPEED,
-            _StrategyType.TPU_SPAWN,
-        )
+    def _choose_and_init_cluster_environment(self):
+        self.cluster_environment = LightningEnvironment()
+        if isinstance(self._cluster_environment, ClusterEnvironment):
+            self.cluster_environment = self._cluster_environment
+        elif self._is_slurm_managing_tasks():
+            rank_zero_info("Multiprocessing is handled by SLURM.")
+            self.cluster_environment = SLURMEnvironment()
+        else:
+            for env_type in (TorchElasticEnvironment, KubeflowEnvironment, LSFEnvironment):
+                if env_type.detect():
+                    self.cluster_environment = env_type()
 
-    @property
-    def use_ddp2(self) -> bool:
-        return self._strategy_type == _StrategyType.DDP2
+    def _is_slurm_managing_tasks(self):
+        """used by choosing cluster enviroment."""
+        if not SLURMEnvironment.detect() or SLURMEnvironment.job_name() == "bash":
+            return False
 
-    @property
-    def use_horovod(self) -> bool:
-        return self._strategy_type == _StrategyType.HOROVOD
+        total_requested_devices = len(self._parallel_devices) * self._num_nodes_flag
+        num_slurm_tasks = int(os.environ["SLURM_NTASKS"], 0)
+        return num_slurm_tasks == total_requested_devices
 
-    @property
-    def use_deepspeed(self) -> bool:
-        return self._strategy_type == _StrategyType.DEEPSPEED
+    def _choose_strategy(self):
```

Comment:
style-wise, you could make this return a str again so it's easier to read & test
```
def _get_strategy_name(self):
  return "..."
```

then it's self._strategy_flag = self._choose_strategy_name()

Reply:
make sense, updated